processing/src/main/java/org/apache/druid/query/spec/SpecificSegmentQueryRunner.java

```diff
@@ -151,12 +150,13 @@ public void close() throws IOException
     );
   }
 
-  private void appendMissingSegment(Map<String, Object> responseContext)
+  private void appendMissingSegment(ResponseContext responseContext)
   {
-    List<SegmentDescriptor> missingSegments = (List<SegmentDescriptor>) responseContext.get(Result.MISSING_SEGMENTS_KEY);
+    List<SegmentDescriptor> missingSegments =
```

Comment:
It's more like a combination of `putIfAbsent` and `computeIfPresent`

Reply:
Let me know if I should introduce these two methods to refactor a method `appendMissingSegment`

Reply:
If `ResponseContext` exposed `computeIfAbsent()` this block might look like
```java
responseContext
    .computeIfAbsent(ResponseContext.CTX_MISSING_SEGMENTS, k -> new ArrayList<>())
    .add(specificSpec.getDescriptor());
```
Unless I miss something.

Generally, yes, I see a value of moving methods into `ResponseContext` because the keys are already defined there. It doesn't seem to me like "god object" or leaky abstraction antipattern, at not least anymore than it is already. But this is not a strong opinion, so feel free to leave things as is if you think otherwise.